django/test/runner.py

```diff
@@ -332,17 +333,23 @@ def run(self, test):
         return result
 
 
-def default_test_processes():
-    """Default number of test processes when using the --parallel option."""
+def parallel_type(string):
+    """Parse value passed to the --parallel option."""
     # The current implementation of the parallel test runner requires
     # multiprocessing to start subprocesses with fork().
     if multiprocessing.get_start_method() != 'fork':
         return 1
-    try:
-        return int(os.environ['DJANGO_TEST_PROCESSES'])
-    except KeyError:
+
+    if string == "auto":
         return multiprocessing.cpu_count()
 
+    try:
+        return int(string)
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            f"{string!r} is not an integer or the string 'auto'"
```

Comment:
Can we add a test for this error message?

Reply:
Done: `test_parallel_bad_string`